src/smart-components/role/RolesTable.tsx

```diff
@@ -86,12 +101,54 @@ const RolesTable: React.FunctionComponent<RolesTableProps> = ({ selectedRole })
     });
   }, [fetchData, page, perPage]);
 
-  const rows = useMemo(() => {
+  const getSortParams = (columnIndex: number): ThProps['sort'] => ({
+    sortBy: {
+      index: sortByIndex,
+      direction,
+      defaultDirection: 'asc',
+    },
+    onSort: (_event, index, direction) => onSort(_event, COLUMNHEADERS[index].key, direction),
+    columnIndex,
+  });
+
+  const sortData = (data: Role[], sortBy: string | undefined, direction: 'asc' | 'desc' | undefined) =>
+    sortBy && direction
+      ? [...data].sort((a, b) =>
+          direction === 'asc'
+            ? a[sortBy as keyof Role] < b[sortBy as keyof Role]
+              ? -1
+              : a[sortBy as keyof Role] > b[sortBy as keyof Role]
+              ? 1
+              : 0
+            : a[sortBy as keyof Role] > b[sortBy as keyof Role]
+            ? -1
+            : a[sortBy as keyof Role] < b[sortBy as keyof Role]
+            ? 1
+            : 0
+        )
+      : data;
```

Comment:
We should not sort data on our side in this case and rather fetch them sorted from the API using `orderBy` param